r_basics_visualize_data/r_basics_visualize_data.md

```diff
@@ -0,0 +1,1064 @@
+</div>
+<div style = "margin: 1rem auto; max-width: 45%; float:left;">
+
+<pre>
+ggplot(data = covid_testing) +
+  geom_histogram(mapping = aes(x = pan_day, fill = result))
+</pre>
+
+</div>
+</div>
+
+In contrast, consider this plot. It's the same as the one you've created at the beginning of the session, except the bars are blue, not black. So the difference is the "fill" aesthetic. But we're not really mapping the fill aesthetic to a variable here, because all bars are the same fill color. They don't represent the values of a column (or variable) of a data frame. Instead, we're setting it to a constant value, the color blue.
+
+![Histogram of Covid tests as a function of `pan_day`.  The bars forming the histogram are blue.](media/blue_histogram.png)
+
+To do this in ggplot, you can still use "fill", but:
+
+* Instead of setting it equal to a column / variable, you'll set it to a color, like "blue" or "#1111FF" (if you use "hex" or "hexidecimal" RGB values)
+* Instead of placing the "fill = " inside `aes()`, you'll move it outside of `aes()`.
+
+This is how you'd get an all-blue histogram:
+
+```
+ggplot(data = covid_testing) +
+  geom_histogram(mapping = aes(x = pan_day), fill = "blue")
+```
+
+Notice **where** the `fill =` appears and what it it set equal to!
+
+Importantly, R knows a lot of different colors by their English names.  To see them, you can use a [great cheat sheet](http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf) that shows the color and the color name, or perhaps [this one](http://derekogle.com/NCGraphing/resources/colors).  For "hex" colors, there are a number of "color pickers" out there, like [Google's](https://g.co/kgs/hv1JrX).
+
+## Working With Geoms
+
+Consider: how are these two plots similar?  How are they different?
+
+<div style = "align-items: center; display: flex;">
+
+<div style = "margin: 1rem; max-width: 45%; float:left;"> ![histogram of covid data](media/mini_histogram.png)
+</div>
+<div style = "margin: 1rem auto; max-width: 45%; float:left;"> ![frequency polygon of covid data](media/mini_freqpoly.png)
+</div>
+</div>
+
+These two graphs share the same data, plotted on the same x and y axes.  What's different? What's different is that on the left, the data is shown as a histogram, and on the right, it's shown as what's called a frequency polygon.
+
+A geom function is a function that, given the data and aesthetic mappings, generates the **geometric object** used to represent the data.
+
+In the next exercise, you're going to work hands-on with geom functions.
+
+### Exercise Time!
+
+In your RStudio environment, you should already have `visualize_exercises.Rmd` open.  If not, please reopen it.  Scroll down to `### Your Turn: Geom functions`, which should be around line 90 or so.
+
+![Lines 91-106 of visualize_exercises.Rmd](media/ex_2_task_1.png)<!-- style = "max-width:700px; border: 1px solid rgb(var(--color-highlight))" -->
```

Comment:
image not appearing

Reply:
fun fact, you gotta enclose text containing underscores in back ticks in alt text!  the more you knoooooow